plugins/UsersManager/API.php

```diff
@@ -795,26 +798,35 @@ public function regenerateTokenAuth($userLogin)
      * Updates a user in the database.
      * Only login and password are required (case when we update the password).
      *
-     * If the password changes and the user has an old token_auth (legacy MD5 format) associated,
-     * the token will be regenerated. This could break a user's API calls.
+     * If password or email changes, it is required to also specify the password of the current user needs to be specified
+     * to confirm this change.
      *
      * @see addUser() for all the parameters
      */
     public function updateUser($userLogin, $password = false, $email = false, $alias = false,
-                               $_isPasswordHashed = false)
+                               $_isPasswordHashed = false, $passwordConfirmation = false)
     {
+        $requirePasswordConfirmation = self::$UPDATE_USER_REQUIRE_PASSWORD_CONFIRMATION;
+        self::$UPDATE_USER_REQUIRE_PASSWORD_CONFIRMATION = true;
+
         Piwik::checkUserHasSuperUserAccessOrIsTheUser($userLogin);
         $this->checkUserIsNotAnonymous($userLogin);
         $this->checkUserExists($userLogin);
 
+        if (!Url::isValidHost()) {
```

Comment:
Moved this from UsersManager controller into here. Might eventually make sense to apply this to more login/user related API methods or even simply in general?

Reply:
Will the Host header always be sent for API requests? If someone curls this API method they would have to send the Host header now, right?

Reply:
just double checked, when no host header is set, it will be considered valid. see https://github.com/matomo-org/matomo/blob/3.8.0-b2/core/Url.php#L210-L215

Reply:
I'll move it back to the UI controller again though I think. seems not needed or much useful here. (not even sure why it is in the controller there)